app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailScreen.kt

```diff
@@ -592,22 +593,22 @@ fun GalleryDetailScreen(args: GalleryDetailScreenArgs, navigator: DestinationsNa
             val downloadTorrentFailed = stringResource(R.string.download_torrent_failure)
             val downloadTorrentStarted = stringResource(R.string.download_torrent_started)
             val noTorrents = stringResource(R.string.no_torrents)
-            var mTorrentList by remember { mutableStateOf<TorrentResult?>(null) }
-            suspend fun showTorrentDialog() {
-                if (mTorrentList == null) {
-                    mTorrentList = bgWork {
-                        withIOContext {
-                            EhEngine.getTorrentList(galleryDetail.torrentUrl!!, gid, token)
-                        }
+            val torrentList = remember {
+                async(Dispatchers.IO) {
+                    runSuspendCatching {
+                        EhEngine.getTorrentList(galleryDetail.torrentUrl!!, gid, token)
```

Comment:
Not sure about this.
Now users who don't use torrents will pay for what they don't need.

Reply:
Make it start lazily?

Reply:
SGTM